From d432100ffcf2f8857ed67b146a50e29d18baaac4 Mon Sep 17 00:00:00 2001
From: Rob Clark <rclark@redhat.com>
Date: Wed, 17 Dec 2014 21:39:20 -0500
Subject: [drm] i915: drop WaSetupGtModeTdRowDispatch:snb

Message-id: <1418852380-13061-70-git-send-email-rclark@redhat.com>
Patchwork-id: 102206
O-Subject: [RHEL7 drm 69/89] drm/i915: drop WaSetupGtModeTdRowDispatch:snb
Bugzilla: 1173317
RH-Acked-by: Jerome Glisse <jglisse@redhat.com>
RH-Acked-by: Dave Airlie <airlied@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1173317
Upstream: since drm-intel-fixes-2014-11-19

commit 2208d655a91f9879bd9a39ff9df05dd668b3512c

Author:     Daniel Vetter <daniel.vetter@ffwll.ch>
AuthorDate: Fri Nov 14 09:25:29 2014 +0100
Commit:     Rob Clark <rclark@redhat.com>
CommitDate: Mon Dec 15 14:18:25 2014 -0500

    drm/i915: drop WaSetupGtModeTdRowDispatch:snb

    This reverts the regressing

    commit 6547fbdbfff62c99e4f7b4f985ff8b3454f33b0f
    Author: Daniel Vetter <daniel.vetter@ffwll.ch>
    Date:   Fri Dec 14 23:38:29 2012 +0100

        drm/i915: Implement WaSetupGtModeTdRowDispatch

    that causes GPU hangs immediately on boot.

    Reported-by: Leo Wolf <jclw@ymail.com>
    Bugzilla: https://bugs.freedesktop.org/show_bug.cgi?id=79996
    Cc: stable@vger.kernel.org (v3.8+)
    Signed-off-by: Daniel Vetter <daniel.vetter@intel.com>
    [Jani: amended the commit message slightly.]
    Signed-off-by: Jani Nikula <jani.nikula@intel.com>

Signed-off-by: Rob Clark <rclark@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/gpu/drm/i915/intel_pm.c b/drivers/gpu/drm/i915/intel_pm.c
index 9d8de8b..bb6c7e1 100644
--- a/drivers/gpu/drm/i915/intel_pm.c
+++ b/drivers/gpu/drm/i915/intel_pm.c
@@ -4945,11 +4945,6 @@ static void gen6_init_clock_gating(struct drm_device *dev)
  I915_WRITE(_3D_CHICKEN,
      _MASKED_BIT_ENABLE(_3D_CHICKEN_HIZ_PLANE_DISABLE_MSAA_4X_SNB));
 
- /* WaSetupGtModeTdRowDispatch:snb */
- if (IS_SNB_GT1(dev))
-  I915_WRITE(GEN6_GT_MODE,
-      _MASKED_BIT_ENABLE(GEN6_TD_FOUR_ROW_DISPATCH_DISABLE));
-
  /* WaDisable_RenderCache_OperationalFlush:snb */
  I915_WRITE(CACHE_MODE_0, _MASKED_BIT_DISABLE(RC_OP_FLUSH_ENABLE));
 
-- 
1.7.1